Documentation for the new DOTNET_GCPath config (#40958)
dotnet/docs: docs/core/runtime-config/garbage-collector.md

## Standalone GC

- Specifies the full path of a GC native library that the runtime loads in place of the default GC implementation. To be secure, this location should be protected from potentially malicious tampering.

| | Setting name | Values | Version introduced |
| - | - | - | - |
| **runtimeconfig.json** | `System.GC.Path` | *string_name* | .NET 9 |
| **Environment variable** | `DOTNET_GCPath` | *string_path* | .NET 9 |

> - Specifies the name of a GC native library that the runtime loads in place of the default GC implementation. This native library needs to reside in the same directory as the assembly that contains your app's `Main` method. If the native module is not found there, then it must reside in the same directory as the .NET runtime (**coreclr.dll** on Windows, **libcoreclr.so** on Linux or **libcoreclr.dylib** on OSX). This configuration setting is ignored if `DOTNET_GCPath` is specified.

Reviewer: ```suggestion
- Specifies the name of a GC native library that the runtime loads in place of the default GC implementation. This native library needs to reside in the same directory as the assembly that contains your app's `Main` method. If the native module is not found there, then it must reside in the same directory as the .NET runtime (*coreclr.dll* on Windows, *libcoreclr.so* on Linux, or *libcoreclr.dylib* on OSX).
- This configuration setting is ignored if `DOTNET_GCPath` is specified.
```

## Standalone GC

> - Specifies the name of a GC native library that the runtime loads in place of the default GC implementation. This native library needs to reside in the same directory as the assembly contained the `Main` method or as the .NET runtime (**coreclr.dll** on Windows, **libcoreclr.so** on Linux).

Reviewer: ```suggestion
- Specifies the name of a GC native library that the runtime loads in place of the default GC implementation. This native library needs to reside in the same directory as the assembly that contains your app's `Main` method or as the .NET runtime (**coreclr.dll** on Windows, **libcoreclr.so** on Linux).
```

> 

Reviewer 2: could we specify the load order here, if I have a GC dll in both where my assembly is and next to coreclr.dll?

| | Setting name | Values | Version introduced |
| - | - | - | - |
| **runtimeconfig.json** | N/A | N/A | N/A |

> | **Environment variable** | `COMPlus_GCName` | *string_name* | .NET Core 2.0 |

Reviewer 2: could you please update this (both can be specified in runtimeconfig)

| **Environment variable** | `DOTNET_GCName` | *string_name* | .NET 6 |

- Specifies the full path of a GC native library that the runtime loads in place of the default GC implementation. To be secure, this location should be protected from potentially malicious tampering.

> 

Reviewer 2: can we also specify what happens if I have both Path and Name specified?

| | Setting name | Values | Version introduced |
| - | - | - | - |
| **runtimeconfig.json** | N/A | N/A | N/A |

> | **Environment variable** | `COMPlus_GCPath` | *string_path* | .NET 9 |

Reviewer: ```suggestion
```

Reviewer: I don't think we need to include this prefix since no-one would have needed to use it previously, correct? I.e. it never existed before the DOTNET prefix version existed?
